src/RequestHeaders.cs

```diff
@@ -18,7 +19,38 @@ public class RequestHeaders : IDictionary<string,IEnumerable<string>> {
     /// </summary>
     /// <param name="headerName">The name of the header to add values to.</param>
     /// <param name="headerValues">The values to add to the header.</param>
-    public void Add(string headerName, params string[] headerValues) {
+    public void Add(string headerName, params string[] headerValues)
+    {
+        if(string.IsNullOrEmpty(headerName))
+            throw new ArgumentNullException(nameof(headerName));
+        if(headerValues == null)
+            throw new ArgumentNullException(nameof(headerValues));
+        if(!headerValues.Any())
+            return;
+        if(_singleValueHeaders.Contains(headerName))
+            _headers[headerName] = new HashSet<string> { headerValues[0] };
+        else if(_headers.TryGetValue(headerName, out var values))
+            foreach(var headerValue in headerValues)
+                values.Add(headerValue);
+        else
+            _headers.Add(headerName, new HashSet<string>(headerValues));
+    }
+    /// <summary>
+    /// Adds values to the header with the specified name if it's not already present
+    /// </summary>
+    /// <param name="headerName">The name of the header to add values to.</param>
+    /// <param name="headerValue">The values to add to the header.</param>
+    public void TryAdd(string headerName, string headerValue)
+    {
+        if(string.IsNullOrEmpty(headerName))
+            throw new ArgumentNullException(nameof(headerName));
+        if(headerValue == null)
+            throw new ArgumentNullException(nameof(headerValue));
+        if(!_headers.ContainsKey(headerName))
+            _headers.Add(headerName, new HashSet<string> { headerValue });
+    }
+    private void AddImpl(string headerName, bool appendIfPresent, params string[] headerValues)
```

Comment:
this method is not used?

Reply:
Looks to be duplicate of the new `Add`. Maybe we should drop one.

Reply:
leftover, sorry for the confusion.

Reply:
removed